File: src/main/java/com/itsci/mju/maebanjumpen/model/SkillType.java
package com.itsci.mju.maebanjumpen.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "skill_type")
@AllArgsConstructor
@NoArgsConstructor
@Data
@JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
public class SkillType {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer skillTypeId;

    @Column(name="skill_type_name", nullable = false)
    private String skillTypeName;

    @Column(name="skill_type_detail", nullable = false)
    private String skillTypeDetail;

    @Column(name="base_price_per_hour", nullable = false) // <-- เพิ่ม field นี้
    private Double basePricePerHour; // ราคาเริ่มต้นต่อชั่วโมงของบริการนี้
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/entity/SkillType.kt
package com.itsci.mju.maebanjumpen.entity

import com.itsci.mju.maebanjumpen.skilltype.dto.SkillTypeDTO
import jakarta.persistence.*

@Entity
@Table(name = "skill_type")
data class SkillType(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = 0,

    @Column(name = "skill_type_name")
    var skillTypeName: String? = null,

    @Column(name = "skill_type_detail")
    var skillTypeDetail: String? = null
){
  fun  toSkillTypeDTO(id: Long?): SkillTypeDTO {
        return SkillTypeDTO(
          id =  id,
          skillTypeName = skillTypeName,
          skillTypeDetail = skillTypeDetail)
    }
}

